### PR TITLE
REF: separate to_time, avoid runtime imports

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/tools/datetimes.py [hunk 1]
```diff
@@ -156,7 +159,7 @@ def _maybe_cache(arg, format, cache, convert_listlike):
 
 def _box_as_indexlike(
     dt_array: ArrayLike, utc: Optional[bool] = None, name: Optional[str] = None
-) -> Union[ABCIndex, ABCDatetimeIndex]:
+) -> Index:
```
Comment (jbrockmendel) [inline pandas/core/tools/datetimes.py:162]: @simonjayhawkins do we have a way of annotating a return type is either an Index or DatetimeIndex, but no other Index subclass?

File: pandas/core/tools/datetimes.py [hunk 2]
```diff
@@ -942,133 +933,3 @@ def calc_with_mask(carg, mask):
         pass
 
     return None
-
```
Comment (jreback) [inline pandas/core/tools/datetimes.py:945]: can you leave to_time here and show a FutureWarning (I know at least ibis is using this)

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]
i think would be ok in exposing `pd.to_time` as well

File: pandas/core/tools/datetimes.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/tools/datetimes.py:945]: updated per request

File: pandas/core/tools/times.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        - If 'raise', then invalid parsing will raise an exception
+        - If 'coerce', then invalid parsing will be set as None
+        - If 'ignore', then invalid parsing will return the input
+
+    Returns
+    -------
+    datetime.time
+    """
+
+    def _convert_listlike(arg, format):
```
Comment (jreback) [inline pandas/core/tools/times.py:42]: in followon can make this a module level function

**Review by jreback — APPROVED** [review]
lgtm. small followon comment, also can you open an issue about exposing ``pd.to_time``

**Comment by jreback** [comment]
can you rebase

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by mroeschke** [comment]
> also can you open an issue about exposing pd.to_time

https://github.com/pandas-dev/pandas/issues/11947

**Comment by jreback** [comment]
once more rebase

File: pandas/core/tools/datetimes.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/tools/datetimes.py:163]: not that i'm aware of. If Index subclasses obey Liskov, shouldn't be an issue. why is this needed?

File: pandas/core/tools/datetimes.py [hunk 4]
```diff
@@ -186,9 +189,9 @@ def _box_as_indexlike(
 
 def _convert_and_box_cache(
     arg: DatetimeScalarOrArrayConvertible,
-    cache_array: ABCSeries,
+    cache_array: "Series",
     name: Optional[str] = None,
-) -> ABCIndexClass:
+) -> "Index":
```
Comment (simonjayhawkins) [inline pandas/core/tools/datetimes.py:194]: quotes no longer needed?

File: pandas/core/tools/datetimes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/tools/datetimes.py:163]: > why is this needed?

Not "needed", but i guess its a more-specific-is-better thing

File: pandas/core/tools/datetimes.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/core/tools/datetimes.py:194]: good catch.  will edit in follow-up

**Comment by jreback** [comment]
thanks!

can you edit the deprecations removal issue for this change as well (not sure if you did)

**Pull request merged**